### PR TITLE
New method for adding tracks at index

### PR DESCRIPTION
**Pull request by dcvz**

### CODE REVIEW

File: kotlin-audio/src/main/java/com/doublesymmetry/kotlinaudio/players/QueuedAudioPlayer.kt [hunk 1]
```diff
@@ -73,11 +73,9 @@ class QueuedAudioPlayer(context: Context, bufferConfig: BufferConfig? = null) :
     /**
      * Add a single item to the queue.
      * @param item The [AudioItem] to add.
-     * @param playWhenReady If this is `true` it will automatically start playback. Default is `true`.
+     * @param playWhenReady If the AudioPlayer has no item loaded, it will load the `item`. If this is `true` it will automatically start playback. Default is `true`.
      */
     fun add(item: AudioItem, playWhenReady: Boolean = true) {
-        exoPlayer.playWhenReady = playWhenReady
```
Comment (mpivchev) [inline kotlin-audio/src/main/java/com/doublesymmetry/kotlinaudio/players/QueuedAudioPlayer.kt:79]: Any reason why this was removed? Also, we still have it as a parameter.

File: kotlin-audio/src/main/java/com/doublesymmetry/kotlinaudio/players/QueuedAudioPlayer.kt [hunk 2]
```diff
@@ -73,11 +73,9 @@ class QueuedAudioPlayer(context: Context, bufferConfig: BufferConfig? = null) :
     /**
      * Add a single item to the queue.
      * @param item The [AudioItem] to add.
-     * @param playWhenReady If this is `true` it will automatically start playback. Default is `true`.
+     * @param playWhenReady If the AudioPlayer has no item loaded, it will load the `item`. If this is `true` it will automatically start playback. Default is `true`.
```
Comment (mpivchev) [inline kotlin-audio/src/main/java/com/doublesymmetry/kotlinaudio/players/QueuedAudioPlayer.kt:76]: The sentence you added doesn't seem related to the `playWhenReady` param? Maybe it should be at the top.

File: kotlin-audio/src/main/java/com/doublesymmetry/kotlinaudio/players/QueuedAudioPlayer.kt (hunk 1 above)
Comment (dcvz) [reply kotlin-audio/src/main/java/com/doublesymmetry/kotlinaudio/players/QueuedAudioPlayer.kt:79]: you're calling it twice, once here and at the end of the method.

File: kotlin-audio/src/main/java/com/doublesymmetry/kotlinaudio/players/QueuedAudioPlayer.kt (hunk 2 above)
Comment (dcvz) [reply kotlin-audio/src/main/java/com/doublesymmetry/kotlinaudio/players/QueuedAudioPlayer.kt:76]: it kind of is, but either way sounds good. I'll update it.

File: kotlin-audio/src/main/java/com/doublesymmetry/kotlinaudio/players/QueuedAudioPlayer.kt (hunk 1 above)
Comment (mpivchev) [reply kotlin-audio/src/main/java/com/doublesymmetry/kotlinaudio/players/QueuedAudioPlayer.kt:79]: Oops 😬

**Review by mpivchev — APPROVED** [review]
LGTM :)
